rest/route/keys_routes.go

```diff
@@ -52,3 +62,54 @@ func (h *keysGetHandler) Execute(ctx context.Context, sc data.Connector) (Respon
 		Result: userKeys,
 	}, nil
 }
+
+type keysPostHandler struct {
+	keyName  string
+	keyValue string
+}
+
+func (h *keysPostHandler) Handler() RequestHandler {
+	return &keysPostHandler{}
+}
+
+func (h *keysPostHandler) ParseAndValidate(ctx context.Context, r *http.Request) error {
+	body := util.NewRequestReader(r)
+	defer body.Close()
+
+	key := model.APIPubKey{}
+	if err := util.ReadJSONInto(body, &key); err != nil {
+		return &rest.APIError{
+			StatusCode: http.StatusBadRequest,
+			Message:    fmt.Sprintf("failed to unmarshal public key: %s", err),
+		}
+	}
+
+	if err := h.validatePublicKey(key); err != nil {
+		return &rest.APIError{
+			StatusCode: http.StatusBadRequest,
+			Message:    fmt.Sprintf("invalid public key: %s", err),
+		}
+	}
+
+	return nil
+}
+
+func (h *keysPostHandler) validatePublicKey(key model.APIPubKey) error {
+	h.keyName = string(key.Name)
+	h.keyValue = string(key.Key)
+	if strings.TrimSpace(h.keyName) == "" || strings.TrimSpace(h.keyValue) == "" {
```

Comment:
I lied and we do some validation in the client. See line 109 of directives.spawn.js. I think we should do some of that here, though if you want to skip that for now it's not too important